internal/app/machined/pkg/controllers/k8s/kubelet_kubeconfig.go

```diff
@@ -0,0 +1,138 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+package k8s
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"os"
+
+	"github.com/cosi-project/runtime/pkg/controller"
+	"github.com/cosi-project/runtime/pkg/resource"
+	"github.com/cosi-project/runtime/pkg/state"
+	"github.com/siderolabs/go-pointer"
+	"go.uber.org/zap"
+	"gopkg.in/yaml.v3"
+
+	"github.com/talos-systems/talos/pkg/conditions"
+	"github.com/talos-systems/talos/pkg/machinery/constants"
+	"github.com/talos-systems/talos/pkg/machinery/resources/config"
+)
+
+// KubeletKubeconfigController renders static pod definitions and manages k8s.KubeconfigStatus.
+type KubeletKubeconfigController struct{}
+
+// Name implements controller.Controller interface.
+func (ctrl *KubeletKubeconfigController) Name() string {
+	return "k8s.KubeletKubeconfigController"
+}
+
+// Inputs implements controller.Controller interface.
+func (ctrl *KubeletKubeconfigController) Inputs() []controller.Input {
+	return []controller.Input{
+		{
+			Namespace: config.NamespaceName,
```

Comment:
https://github.com/AlekSi/talos/blob/2a2eba4705e0fb8ba057a753028173b79dd53af5/pkg/machinery/resources/secrets/kubelet.go#L26-L34 

this is a proper resource for the endpoint, so that we don't depend on machine config everywhere

Reply:
Do you suggest adding it as a new field to it? Because right now it doesn't have it.

Reply:
Endpoint is the first field there unless I'm missing something

Reply:
oh right, totally missed that one 🤦